float32 robotx
float32 roboty
float32 robottheta
float32 robotcos
float32 robotsin
